addons/point_of_sale/point_of_sale.py

```diff
@@ -1463,6 +1466,16 @@ def create_from_ui(self, cr, uid, partner, context=None):
         
         return partner_id
 
+class account_move(osv.osv):
+    _inherit = 'account.move'
 
+    def create(self, cr, uid, vals, context=None):
+        if 'last_order_date' in context:
+            account_period_obj = self.pool.get('account.period')
+            period_id = account_period_obj.find(cr, uid, dt=context['last_order_date'], context=context)
+            result = super(account_move, self).create(cr, uid, dict(vals, date=context['last_order_date'], period_id=period_id[0]), context)
+        else:
+            result = super(account_move, self).create(cr, uid, vals, context)
+        return result
```

Comment:
No need at all either. Always try to never override create/write, that's not the best practice :).